assets/src/hooks/useRouteFilter.ts

```diff
@@ -0,0 +1,42 @@
+import { useState } from "react"
+
+type FilterType = "id"
+
+export interface RouteFilter {
+  filterType: FilterType
+  filterText: string
+  handleTypeChange: (event: React.FormEvent<HTMLSelectElement>) => void
+  handleTextInput: (event: React.FormEvent<HTMLInputElement>) => void
+  clearTextInput: () => void
+}
+
+const isFilterType = (str: string): str is FilterType => str === "id"
```

Comment:
Since we only have one filter type now, probably shouldn't set up all the infrastructure for switching types yet, and just assume that the text is for matching ids. If you think it'd be a lot of work to rip it out and add it back in later, then I guess it's fine, but I wouldn't be surprised if the requirements for filtering by something else like stop are so complicated that a lot of this logic would need to change anyway.

Reply:
Yeah, this is something I might think about handling later, but in this case I wanted to show the type picker in the view in order to get the layout correct, and it seemed more important to have the structure backing the view rather than leaving it "floating". But this is also why I was expecting the filter function to have logic that varies based on the type.